compiler/damlc/daml-opts/daml-opts/DA/Daml/Options.hs

```diff
@@ -83,10 +96,18 @@ getDamlGhcSession :: Options -> IO (FilePath -> Action HscEnvEq)
 getDamlGhcSession options@Options{..} = do
     findProjectRoot <- memoIO findProjectRoot
     getSession <- memoIO $ \mbProjectRoot -> do
+        let base = mkBaseUnits optMbPackageName
+        optPackageImports <-
+          if getInferDependantPackages optInferDependantPackages
+          then do
+            more <- dependantUnitsFromDamlYaml mbProjectRoot
+            pure $ optPackageImports ++ map mkPackageFlag (base ++ more)
```

Comment:
I think it probably makes more sense to put `optPackageImports` at the end so you can infer the inferred flags.

Reply:
I did as you suggest. I assume this is because later options can override/extend earlier ones.